lua/neogen/generator.lua

```diff
@@ -0,0 +1,229 @@
+    return language.locator({root = tree, current = ts_utils.get_node_at_cursor(0)}, language.parent[typ])
+end
+
+---Generates the prefix according to `template` options.
+---Prefix is generated with an offset (currently spaces) repetition of `n` times.
+---If `template.use_default_comment` is not set to false, the `commentstring` is added
+---@param template table
+---@param commentstring string
+---@param n integer
+---@return string
+local function prefix_generator(template, commentstring, n)
+    local prefix = (" "):rep(n)
+
+    -- Do not append the comment string if not wanted
+    if template.use_default_comment ~= false then
+        prefix = prefix .. commentstring
+    end
+    return prefix
+end
+
+local function get_place_pos(parent, position, append, typ)
+    local row, col
+    -- You can use a custom position placement
+    if position and type(position) == "function" then
+        row, col = position(parent, typ)
+    end
+
+    -- If the custom placement does not return the correct row and cols, default to the node range
+    -- Same if there is no custom placement
+    if not row and not col then
+        -- Because the file type is always at top
+        if typ == "file" then
+            row, col = 0, 0
+        else
+            row, col = ts_utils.get_node_range(parent)
+        end
+    end
+
+    append = append or {}
+
+    if append.position == "after" and not vim.tbl_contains(append.disabled or {}, typ) then
+        local child_node = nodes:matching_child_nodes(parent, append.child_name)[1]
+        if not child_node and append.fallback then
+            local fallback = nodes:matching_child_nodes(parent, append.fallback)[1]
+            if fallback then
+                row, col = fallback:range()
+            end
+        else
+            row, col = child_node:range()
+        end
+    end
+
+    return row, col
+end
+
+---Uses the provided template to format the annotations with data found by the granulator
+---@param parent userdata the node used to generate the annotations
+---@param data table the data from the granulator, which is a set of [type] = results
+---@param template table a template from the configuration
+---@param jump_text string jump_text from the default or language configuration
+---@param required_type string
+---@return table { line, content }, with line being the line to append the content
+local function generate_content(parent, data, template, required_type, jump_text)
+    assert(template, "Need a template table")
+    local row, col = get_place_pos(parent, template.position, template.append, required_type)
+
+    local commentstring = vim.trim(vim.bo.commentstring:format(""))
+    local generated_template
+    if not template.annotation_convention then
+        -- Default template
+        generated_template = {{nil, ""}, {"name", " @Summary "}, {"parameters", " @Param "}, {"return", " @Return "}}
```

Comment:
I don't think this line is required anymore. But we could replace with a ERROR notify